app/models/mortgage_calculator/land_transaction_tax.rb

```diff
@@ -13,10 +13,26 @@ def self.i18n_scope
       { threshold: nil, rate: 12 }
     ].freeze
 
+    %i[price tax_due total_due].each do |field|
+      define_method "#{field}_formatted" do
+        number_to_currency(
+          public_send(field).presence || 0, unit: '', precision: 2
+        )
+      end
+    end
```

Comment:
I know that depends on the taste but I don't know about this meta programming here. Can we not have a private method contain the logic and the definition of this three methods calling the private method passing the field as parameter? Will not read better?

Reply:
Agree with @tomas-stefano's comment. This method was currently in place and I copy/pasted to see if the approach of amending the method in the child model was appropriate. Happy to do the refactoring Tomas suggests.

Reply:
@tomas-stefano please would you comment on the general approach